### PR TITLE
rename rpma engine and add custom engine options

### PR DESCRIPTION
**Pull request by lukaszstolarczuk**
This change is [Reviewable](https://reviewable.io/reviews/janekmi/fio/1)

### CODE REVIEW

**Review by janekmi — CHANGES_REQUESTED** [review]
> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/janekmi/fio/1)* status: all files reviewed, 8 unresolved discussions (waiting on @lukaszstolarczuk)

___
*[engines/librpma.c, line 2 at r1](https://reviewable.io/reviews/janekmi/fio/1#-MHC2x4d2_B5kIyc6E-i:-MHC2x4d2_B5kIyc6E-j:bpid1el) ([raw file](https://github.com/janekmi/fio/blob/2b9b4cb71373a09f1c504940234a4c4ae00fb148/engines/librpma.c#L2)):*
> ```C
> /*
>  * librpma I/O engine - client part
> ```

I want to keep both parts in the same .c file. So we do not need to make any further changes to the build system.
___
*[engines/librpma.c, line 66 at r1](https://reviewable.io/reviews/janekmi/fio/1#-MHC3hdT2ggUufAUorkV:-MHC3hdT2ggUufAUorkW:b-s8visb) ([raw file](https://github.com/janekmi/fio/blob/2b9b4cb71373a09f1c504940234a4c4ae00fb148/engines/librpma.c#L66)):*
> ```C
> struct fio_librpmaio_options {
> 	struct thread_data *td;
> 	unsigned int server_port;
> ```

char *
___
*[engines/librpma.c, line 70 at r1](https://reviewable.io/reviews/janekmi/fio/1#-MHC42mJ0TTd2XF_lqZR:-MHC42mJ0TTd2XF_lqZS:b25toj3) ([raw file](https://github.com/janekmi/fio/blob/2b9b4cb71373a09f1c504940234a4c4ae00fb148/engines/librpma.c#L70)):*
> ```C
> };
> 
> static struct fio_option options[] = {
> ```

client_options
___
*[engines/librpma.c, line 84 at r1](https://reviewable.io/reviews/janekmi/fio/1#-MHC3t2l6TH878R2dVum:-MHC3t2l6TH878R2dVun:b3kqx) ([raw file](https://github.com/janekmi/fio/blob/2b9b4cb71373a09f1c504940234a4c4ae00fb148/engines/librpma.c#L84)):*
> ```C
> 		.name	= "server_port",
> 		.lname	= "librpma engine server port",
> 		.type	= FIO_OPT_INT,
> ```

.
___
*[engines/librpma.c, line 97 at r1](https://reviewable.io/reviews/janekmi/fio/1#-MHC4IqN03U6ozCeLcS_:-MHC3yWOBC5OxYMy0xF9:b-v399ic) ([raw file](https://github.com/janekmi/fio/blob/2b9b4cb71373a09f1c504940234a4c4ae00fb148/engines/librpma.c#L97)):*
> ```C
> };
> 
> /* XXX: for server version of the engine
> ```

We will keep both in the same .c file.
___
*[engines/librpma.c, line 104 at r1](https://reviewable.io/reviews/janekmi/fio/1#-MHC45Pj0aG0nBP-Bung:-MHC45Pj0aG0nBP-Bunh:b2lr5k7) ([raw file](https://github.com/janekmi/fio/blob/2b9b4cb71373a09f1c504940234a4c4ae00fb148/engines/librpma.c#L104)):*
> ```C
> };
> 
> static struct fio_option options[] = {
> ```

server_options
___
*[examples/librpma-server.fio, line 5 at r1](https://reviewable.io/reviews/janekmi/fio/1#-MHC5Aby2cuYy71ZeiLd:-MHC5Aby2cuYy71ZeiLe:b3kqx) ([raw file](https://github.com/janekmi/fio/blob/2b9b4cb71373a09f1c504940234a4c4ae00fb148/examples/librpma-server.fio#L5)):*
> ```
> ioengine=librpma_server
> listen_ip=192.168.13.207
> listen_port=7204
> ```

.
___
*[examples/librpma-client.fio, line 5 at r1](https://reviewable.io/reviews/janekmi/fio/1#-MHC4eKO3aATuraBw93Y:-MHC4eKO3aATuraBw93Z:b-kcx2gs) ([raw file](https://github.com/janekmi/fio/blob/2b9b4cb71373a09f1c504940234a4c4ae00fb148/examples/librpma-client.fio#L5)):*
> ```
> ioengine=librpma
> server_ip=192.168.13.207
> server_port=7204
> ```

Please see https://github.com/janekmi/fio/blob/master/examples/rdmaio-client.fio e.g.:

```
hostname=[hostname]
port=[port]
```

etc.

File: engines/librpma.c [hunk 1]
```diff
@@ -59,94 +61,73 @@ enum librpma_io_mode {
 	FIO_RDMA_CHA_RECV
 };
 
-struct librpmaio_options {
+struct fio_librpmaio_options {
 	struct thread_data *td;
-	unsigned int port;
-	enum librpma_io_mode verb;
-	char *bindname;
+	unsigned int server_port;
```
Comment (kmalek-priv) [inline engines/librpma.c]: Can we use char* for port variable? As I see rpma (eg. rpma_conn_req_new) is using a port as char*.

**Review by lukaszstolarczuk — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/janekmi/fio/1)* status: all files reviewed, 4 unresolved discussions (waiting on @janekmi)

___
*[engines/librpma.c, line 2 at r1](https://reviewable.io/reviews/janekmi/fio/1#-MHC2x4d2_B5kIyc6E-i:-MHFSBM09Tte2NZd4z5I:b-896fix) ([raw file](https://github.com/janekmi/fio/blob/2b9b4cb71373a09f1c504940234a4c4ae00fb148/engines/librpma.c#L2)):*
<details><summary><i>Previously, janekmi (Jan Michalski) wrote…</i></summary><blockquote>

I want to keep both parts in the same .c file. So we do not need to make any further changes to the build system.
</blockquote></details>

Done.
___
*[engines/librpma.c, line 66 at r1](https://reviewable.io/reviews/janekmi/fio/1#-MHC3hdT2ggUufAUorkV:-MHFSED5E9beAr2SyWSj:b-896fix) ([raw file](https://github.com/janekmi/fio/blob/2b9b4cb71373a09f1c504940234a4c4ae00fb148/engines/librpma.c#L66)):*
<details><summary><i>Previously, janekmi (Jan Michalski) wrote…</i></summary><blockquote>

char *
</blockquote></details>

Done.
___
*[engines/librpma.c, line 70 at r1](https://reviewable.io/reviews/janekmi/fio/1#-MHC42mJ0TTd2XF_lqZR:-MHFSMJu92xqZuRfhBUt:b-896fix) ([raw file](https://github.com/janekmi/fio/blob/2b9b4cb71373a09f1c504940234a4c4ae00fb148/engines/librpma.c#L70)):*
<details><summary><i>Previously, janekmi (Jan Michalski) wrote…</i></summary><blockquote>

client_options
</blockquote></details>

Done.
___
*[engines/librpma.c, line 84 at r1](https://reviewable.io/reviews/janekmi/fio/1#-MHC3t2l6TH878R2dVum:-MHFSJVzDf7Ac8SHCSJt:b-896fix) ([raw file](https://github.com/janekmi/fio/blob/2b9b4cb71373a09f1c504940234a4c4ae00fb148/engines/librpma.c#L84)):*
<details><summary><i>Previously, janekmi (Jan Michalski) wrote…</i></summary><blockquote>

.
</blockquote></details>

Done.
___
*[engines/librpma.c, line 97 at r1](https://reviewable.io/reviews/janekmi/fio/1#-MHC4IqN03U6ozCeLcS_:-MHFSQ9t44I7yDo6V51b:b-896fix) ([raw file](https://github.com/janekmi/fio/blob/2b9b4cb71373a09f1c504940234a4c4ae00fb148/engines/librpma.c#L97)):*
<details><summary><i>Previously, janekmi (Jan Michalski) wrote…</i></summary><blockquote>

We will keep both in the same .c file.
</blockquote></details>

Done.
___
*[engines/librpma.c, line 104 at r1](https://reviewable.io/reviews/janekmi/fio/1#-MHC45Pj0aG0nBP-Bung:-MHFSbshDHZlqtXHglWX:b-896fix) ([raw file](https://github.com/janekmi/fio/blob/2b9b4cb71373a09f1c504940234a4c4ae00fb148/engines/librpma.c#L104)):*
<details><summary><i>Previously, janekmi (Jan Michalski) wrote…</i></summary><blockquote>

server_options
</blockquote></details>

Done.
___
*[examples/librpma-server.fio, line 5 at r1](https://reviewable.io/reviews/janekmi/fio/1#-MHC5Aby2cuYy71ZeiLd:-MHFTRGZ4ztPuriMtzNS:b-896fix) ([raw file](https://github.com/janekmi/fio/blob/2b9b4cb71373a09f1c504940234a4c4ae00fb148/examples/librpma-server.fio#L5)):*
<details><summary><i>Previously, janekmi (Jan Michalski) wrote…</i></summary><blockquote>

.
</blockquote></details>

Done.
___
*[examples/librpma-client.fio, line 5 at r1](https://reviewable.io/reviews/janekmi/fio/1#-MHC4eKO3aATuraBw93Y:-MHFTQi_9foBWBOAKvnn:b-896fix) ([raw file](https://github.com/janekmi/fio/blob/2b9b4cb71373a09f1c504940234a4c4ae00fb148/examples/librpma-client.fio#L5)):*
<details><summary><i>Previously, janekmi (Jan Michalski) wrote…</i></summary><blockquote>

Please see https://github.com/janekmi/fio/blob/master/examples/rdmaio-client.fio e.g.:

```
hostname=[hostname]
port=[port]
```

etc.
</blockquote></details>

Done.

**Review by janekmi — APPROVED** [review]
:lgtm:

> Reviewed 3 of 3 files at r2.
> *[Reviewable](https://reviewable.io/reviews/janekmi/fio/1#-:-MHFauuW--mju2_AMFW_:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved

**Review by janekmi — APPROVED** [review]
> Reviewed 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/janekmi/fio/1)* status: :shipit: complete! all files reviewed, all discussions resolved

**Review by janekmi — APPROVED** [review]
> Reviewed 1 of 1 files at r4.
> *[Reviewable](https://reviewable.io/reviews/janekmi/fio/1)* status: :shipit: complete! all files reviewed, all discussions resolved

**Pull request merged**